### PR TITLE
FIX: handle 1D array in gaussian filter

### PR DESCRIPTION
**Pull request by sciunto**
## Description

See #5195 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/filters/tests/test_gaussian.py [hunk 1]
```diff
@@ -19,6 +19,11 @@ def test_negative_sigma():
                  sigma=np.asarray([-1.0, 1.0]))
 
 
+def test_1D():
+    x = np.arange(50) / 50
+    y = filters.gaussian(x, sigma=(1,))
```
Comment (rfezzani) [inline skimage/filters/tests/test_gaussian.py:24]:
```suggestion
    y = gaussian(x, sigma=(1,))
```

**Review by rfezzani — COMMENTED** [review]
Thank you @sciunto, I think that the beginning of the `gaussian` function and may be the `_guess_spatial_dimensions` error message must be refactored. But I am not sure that an error have to be raised since it is filtered by a `try ... Except ...` block...

**Comment by sciunto** [comment]
> Thank you @sciunto, I think that the beginning of the `gaussian` function and may be the `_guess_spatial_dimensions` error message must be refactored. But I am not sure that an error have to be raised since it is filtered by a `try ... Except ...` block...

I would even wonder whether keeping a guess is the right thing. I thought we want to avoid as much as possible magic. Other opinions @jni @emmanuelle ?

**Comment by grlee77** [comment]
> I would even wonder whether keeping a guess is the right thing. I thought we want to avoid as much as possible magic. Other opinions @jni @emmanuelle ?

We have mostly gotten rid of this type of guessing elsewhere (this function is one exception noted in the list at the top of #2613). It looks like @jni and I discussed it here: https://github.com/scikit-image/scikit-image/issues/2613#issuecomment-656720229 and decided to address it as part of #4294. Working on that issue is one of my near-term priorities for the CZI grant, so we should be able to make progress soon.

File: skimage/filters/_gaussian.py [hunk 2]
```diff
@@ -143,8 +143,10 @@ def _guess_spatial_dimensions(image):
     Raises
     ------
     ValueError
-        If the image array has less than two or more than four dimensions.
+        If the image array has less more than four dimensions.
```
Comment (grlee77) [inline skimage/filters/_gaussian.py:146]:
```suggestion
        If the image array has less than four dimensions.
```

**Review by grlee77 — APPROVED** [review]
I vote for merging this as-is and then deal with removing `_guess_spatial_dimensions` separately

File: skimage/filters/tests/test_gaussian.py [hunk 3]
```diff
@@ -19,6 +19,11 @@ def test_negative_sigma():
                  sigma=np.asarray([-1.0, 1.0]))
 
 
+def test_1D():
```
Comment (rfezzani) [inline skimage/filters/tests/test_gaussian.py:22]: What is this test supposed to test? Isn't it equivalent to sample from #5195? It doesn't fail with the previous version...

**Review by rfezzani — APPROVED** [review]
Thank you @sciunto, waiting for the green button before merging :wink:

**Comment by jni** [comment]
Failure is expected from #5211, so I'll merge now. Thanks everyone!

**Comment by stefanv** [comment]
Thanks for fixing this @sciunto!

The new docstring does not look right: `ValueError` "If the image array has less than four dimensions.".

**Pull request merged**